DM-45576: Start a nightly cronjob to post a brief summary

hsinfang opened the pull request.

hsinfang committed:
Install lsst-efd-client in the job container

hsinfang committed:
Clarify pipeline output accounting and improve wording

The number for single-frame-only processing as calculated here
can become wrong as the implementations change, but useful for
the time being.

Using the existence of initial_pvi as the ProcessCcd success
is confusing because calibrateImage can fail after generating
initial_pvi.

Using the last product of calibrateImage is not ideal either,
but a little closer.

Send message only if there are on-sky exposures.
can_see_sky started to be a useable criteria starting w_2024_33.
Using can_see_sky or science_program typically give the same results.

hsinfang committed:
Reformat with black

hsinfang committed:
Use queryDimensionRecords instead of queryDatasets

Switch as this is just to get raw-image metadata.

hsinfang committed:
Report based on Sasquatch and Loki

hsinfang committed:
Update for the rubin-env 9 stack images

Started at w_2024_32

hsinfang committed:
Update survey name from AUXTEL_PHOTO_IMAGING to BLOCK-306

The name change happened on 2024-09-04

ctslater commented on scripts/utils.py:
```diff
@@ -0,0 +1,215 @@
+        f'{{app="vcluster--usdf-prompt-processing",pod=~"{pod_name}-.+"}} {search_string}',
+    ]
+
+    print("Command:", command)
```
Left over debugging?

hsinfang replied:
Yes. Thank you for catching it.

ctslater commented on scripts/prompt_processing_summary.py:
```diff
@@ -1,10 +1,13 @@
 from datetime import date, timedelta
 import requests
 
+from utils import *
```
"utils" is a bit of a naming pet-peeve of mine, but I understand that the functions are somewhat varied. Trying to think of some other name options, maybe "db_queries" or "queries"? If nothing else comes to mind then it's ok to just keep it as utils. I'd also list the functions instead of `*`.

hsinfang replied:
"queries" is a good name; thanks for the idea. 

Also import specific functions rather than `*` now.

ctslater commented on scripts/prompt_processing_summary.py:
```diff
@@ -44,38 +63,98 @@ def make_summary_message(day_obs):
+                    output_lines.append(f"  - {counts} to be investigated.")
+
+    output_lines.append(
+        f"<https://usdf-rsp-dev.slac.stanford.edu/times-square/github/lsst-sqre/times-square-usdf/prompt-processing/groups?date={day_obs}&instrument=LATISS&survey={survey}&mode=DEBUG&ts_hide_code=1|Timing plots>"
```
What does `&mode=DEBUG` do?

hsinfang replied:
In that Times Square notebook, written before we started making this reporting tool, I wanted a "report" mode versus a "debug" mode. The former is brief and meant for people who don't care about PP internals.  The latter has a lot more details which the PP team can use to know what might have gone wrong.  Now we have this reporting tool, I probably can go back and think again whether it makes sense.

ctslater reviewed (approved):
Nice additions to the script. I left just some minor comments.

hsinfang committed:
Switch to the new Butler query system

Starting w_2024_37, the new Butler query system becomes public.

Using explain=False so not to raise EmptyQueryResultError.

hsinfang committed:
Constrain the survey in all queries

Now we also process some test surveys, constrain this report
to the science survey only.
